lib/galaxy/jobs/__init__.py

```diff
@@ -1510,7 +1510,7 @@ def pause(self, job=None, message=None):
             job = self.get_job()
         if message is None:
             message = "Execution of this dataset's job is paused"
-        if job.state == job.states.NEW:
+        if job.state in (job.states.NEW, job.states.QUEUED):
```

Comment:
Can a queued job that is already handed over to the job scheduler be paused?

Reply:
Not really, but the "over-quota pause check" is done now during the `enqueue` process and the state at this point is already `queued`.
I'll see if I can delay the status change without other consequences until this check has been made :+1:

Reply:
I ended up refactoring a bit more than I would have liked for a fix, but it seems reasonable since otherwise the logging will be inconsistent and tell that the job was dispatched when it wasn't.

Reply:
> Can a queued job that is already handed over to the job scheduler be paused?

Yes it can, at any point in its lifetime. I don't know if that means some of the refactoring can be undone ?

Reply:
Ok, Thanks for the clarification! I will drop the additional refactoring commits.

Reply:
@mvdbeek but does that mean that the job scheduler needs to remove the job from the queue? If so should we not avoid that? Just want to understand how this works :)

Reply:
paused jobs won't be picked up by the handler loop, they're excluded in the query.